src/integrators/wrapper_integrator.hpp

```diff
@@ -0,0 +1,86 @@
+// Copyright (c) 2019-2020, Lawrence Livermore National Security, LLC and
+// other Serac Project Developers. See the top-level LICENSE file for
+// details.
+//
+// SPDX-License-Identifier: (BSD-3-Clause)
+
+#ifndef WRAPPER_INTEGRATOR_HPP
+#define WRAPPER_INTEGRATOR_HPP
+
+#include "mfem.hpp"
+
+/// A class to convert linearform integrators into a nonlinear residual-based one
+class LinearToNonlinearFormIntegrator : public mfem::NonlinearFormIntegrator
+{
+public:
+  /**
+     \brief Recasts, A(u) = F as R(u) = A(u) - F
+
+  \param[in] f A LinearFormIntegrator
+   */
+  LinearToNonlinearFormIntegrator( mfem::LinearFormIntegrator & f, mfem::ParFiniteElementSpace *trial_fes);
+
+  /// Compute the residual vector => -F
+  virtual void AssembleElementVector( const mfem::FiniteElement &el, mfem::ElementTransformation &Tr,
+                                      const mfem::Vector &elfun, mfem::Vector & elvect);
+
+  /// Compute the tangent matrix = 0
+  virtual void AssembleElementGrad( const mfem::FiniteElement &el, mfem::ElementTransformation &Tr,
+                                    const mfem::Vector &elfun, mfem::DenseMatrix &elmat);
+
+private:
+  mfem::LinearFormIntegrator & m_f;
```

Comment:
Are you able to use std::shared_ptr here?

Reply:
Good point. I do think std::shared_ptr is better here because it would further guarantee some other piece of code can't pull the rug out from this class.